runtime/parachains/src/util.rs

```diff
@@ -39,3 +40,8 @@ pub fn make_persisted_validation_data<T: paras::Config + hrmp::Config>(
 		max_pov_size: config.max_pov_size,
 	})
 }
+
+/// Take the active subset of a set containing all validators.
+pub fn take_active_subset<T: Clone>(active_validators: &[ValidatorIndex], set: &[T]) -> Vec<T> {
+	active_validators.iter().filter_map(|i| set.get(i.0 as usize)).cloned().collect()
```

Comment:
Panic in the runtime means taking down the entire network with no avenue to recovery, even via governance, as it would be impossible to author any further blocks. I really don't want that.

Reply:
Ok, I can understand that. But just silently ignoring a logic error, makes it very easy to 1 make that logic error reach a production system, 2 make it potentially incredibly hard to debug the problem in the production system.

Uh! I have the solution - what about a debug_assert? Won't trigger in production, but will still help catch errors early.

Reply:
We never run runtimes in debug mode anyway. I think we would notice quite quickly if the number of validators, when supposedly constrained to `max` is less than `max`.